### ISSUE TITLE
Sentinel crashes mogwaid

### ISSUE THREAD

**Issue by mooleshacat** [mogwaicoin/mogwai-sentinel#3]
***Sentinel is crashing mogwaid when run...***

I have full mn sync before run. I am running as mogwai user inside of sentinel directory.

> mogwai@mn0:~/mogwai-sentinel$ cd /home/mogwai/mogwai-sentinel ; SENTINEL_DEBUG=1 ./venv/bin/python bin/sentinel.py
> 2018-10-15 16:05:51 [info]: SCHEMA_VERSION (code) = [20170111-1]
> 2018-10-15 16:05:51 [info]: DB_SCHEMA_VERSION = [20170111-1]
> ('SELECT "t1"."id", "t1"."name", "t1"."value", "t1"."created_at", "t1"."updated_at" FROM "settings" AS t1 WHERE ("t1"."name" = ?) LIMIT 1 OFFSET 0', [u'__transient_NEXT_SENTINEL_CHECK_AT'])
> 2018-10-15 16:05:51 current_time = 1539619551
> 2018-10-15 16:05:51 next_run_time = 0
> 2018-10-15 16:05:51 Delay of [56] seconds for cron minute offset
> 
> ('SELECT "t1"."id", "t1"."name", "t1"."value", "t1"."created_at", "t1"."updated_at" FROM "settings" AS t1 WHERE ("t1"."name" = ?) LIMIT 1 OFFSET 0', [u'__transient_NEXT_SENTINEL_CHECK_AT'])
> ('SELECT "t1"."id", "t1"."parent_id", "t1"."object_creation_time", "t1"."object_hash", "t1"."object_parent_hash", "t1"."object_type", "t1"."object_revision", "t1"."object_fee_tx", "t1"."yes_count", "t1"."no_count", "t1"."abstain_count", "t1"."absolute_yes_count" FROM "governance_objects" AS t1', [])
> 2018-10-15 16:06:47 in sentinel_ping
> 2018-10-15 16:06:47 leaving sentinel_ping
> ('SELECT "t1"."id", "t1"."governance_object_id", "t1"."name", "t1"."url", "t1"."start_epoch", "t1"."end_epoch", "t1"."payment_address", "t1"."payment_amount", "t1"."object_hash" FROM "proposals" AS t1', [])
> ('SELECT Count(*) FROM "superblocks" AS t1 INNER JOIN "governance_objects" AS t5 ON ("t1"."governance_object_id" = "t5"."id") INNER JOIN "votes" AS t2 ON ("t5"."id" = "t2"."governance_object_id") INNER JOIN "signals" AS t3 ON ("t2"."signal_id" = "t3"."id") INNER JOIN "outcomes" AS t4 ON ("t2"."outcome_id" = "t4"."id") WHERE ((("t1"."event_block_height" = ?) AND ("t2"."signal_id" = ?)) AND ("t2"."outcome_id" = ?))', [75600, 1, 1])
> Traceback (most recent call last):
>   File "bin/sentinel.py", line 265, in <module>
>     main()
>   File "bin/sentinel.py", line 221, in main
>     attempt_superblock_creation(mogwaid)
>   File "bin/sentinel.py", line 109, in attempt_superblock_creation
>     proposals = Proposal.approved_and_ranked(proposal_quorum=mogwaid.governance_quorum(), next_superblock_max_budget=mogwaid.next_superblock_max_budget())
>   File "lib/mogwaid.py", line 136, in next_superblock_max_budget
>     next_allocation = self.get_superblock_budget_allocation(next_superblock_height)
>   File "lib/mogwaid.py", line 126, in get_superblock_budget_allocation
>     return Decimal(self.rpc_command('getsuperblockbudget', height))
>   File "lib/mogwaid.py", line 42, in rpc_command
>     return self.rpc_connection.__getattr__(params[0])(*params[1:])
>   File "/home/mogwai/mogwai-sentinel/venv/local/lib/python2.7/site-packages/bitcoinrpc/authproxy.py", line 139, in __call__
>     response = self._get_response()
>   File "/home/mogwai/mogwai-sentinel/venv/local/lib/python2.7/site-packages/bitcoinrpc/authproxy.py", line 179, in _get_response
>     http_response = self.__conn.getresponse()
>   File "/usr/lib/python2.7/httplib.py", line 1089, in getresponse
>     response.begin()
>   File "/usr/lib/python2.7/httplib.py", line 444, in begin
>     version, status, reason = self._read_status()
>   File "/usr/lib/python2.7/httplib.py", line 408, in _read_status
>     raise BadStatusLine(line)
> httplib.BadStatusLine: ''
> ('SELECT "t1"."id", "t1"."name", "t1"."value", "t1"."created_at", "t1"."updated_at" FROM "settings" AS t1 WHERE ("t1"."name" = ?) LIMIT 1 OFFSET 0', [u'__transient_SENTINEL_RUNNING'])
> ('DELETE FROM "settings" WHERE ("id" = ?)', [2])
> mogwai@mn0:~/mogwai-sentinel$ 
> mogwai@mn0:~/mogwai-sentinel$ mogwai-cli mnsync status
> error: couldn't connect to server
> mogwai@mn0:~/mogwai-sentinel$

***Any idea why?***

***Thanks***

**Comment by darkfriend77** [comment]
I have an idea ... it might be linked to 

`        // Deployment of DIP0001
        consensus.vDeployments[Consensus::DEPLOYMENT_DIP0001].bit = 1;
        consensus.vDeployments[Consensus::DEPLOYMENT_DIP0001].nStartTime = 1508025600; // Oct 15th, 2017
        consensus.vDeployments[Consensus::DEPLOYMENT_DIP0001].nTimeout = 1539561600;   // Oct 15th, 2018
        consensus.vDeployments[Consensus::DEPLOYMENT_DIP0001].nWindowSize = 4032;
        consensus.vDeployments[Consensus::DEPLOYMENT_DIP0001].nThreshold = 3226; // 80% of 4032
`

**Comment by mooleshacat** [comment]
Thanks for your reply @darkfriend77 , no one else seems to be answering.

I fixed it somehow during a server move... Something with reinstalling the python-pip and another python package (forget which one) and rebuilding the python venv as per installation instructions in the README.md.

Also I noticed one of my two MOG MN's was pointing to the other one's sentinel path (not it's own)

***Fix:***

**1)** Make sure all packages are installed (python-pip, etc - check README.md)
**2)** Make sure if you have multiple users for one coin that the paths are correct.
**3)** Fully reinstall sentinel from the beginning (if all else fails)
